### DISCUSSION TITLE
Modem e mancanza alimentazione

### DISCUSSION THREAD

**Discussion by fraygon** [indomus/forum#3235] [Networking]
Ciao a tutti. Vi volevo chiedere se mi sapete dare qualche risposta riguardo a questo fatto: è la prima volta che mi succede e mi ha spiazzato. Ho HA su RPI 4 collegato via lan al modem fastgate (non quello a cubo ma il penultimo tipo). Ho acquistato un inverter per alimentare entrambi e il tutto ha sempre funzionato quando mi scatta il differenziale di casa. 
Ieri gli amici di terna hanno tolto l'alimentazione a tutta la zona e il modem non aveva più collegamento con la rete (strano) e il mio H.A. non è stato più raggiungibile nè in locale nè in remoto: funzionava perchè ho alcuni led che mi comunicano varie attività ma io non vedevo niente dall'app companion sul cellulare.
Dopo tutta questa papardella vi chiedo se è la stessa cosa per voi e se ho sbagliato qualcosa nella installazione perchè credevo che in locale avrei sempre visto HA, anche senza linea esterna. 
Spero di essermi spiegato. Grazie

**Comment by Scialla** [comment]
in locale, se collegato alla tua wifi, deve funzionare, se hai impostato correttamente in `configuration.yaml` e se hai impostato correttamente sulla app companion

```
homeassistant:
  internal_url: http://tuo_ip:8123
  external_url: https://tuo_fqdn
  ```

**Reply by fraygon** [reply]
Allora è questo il problema. in locale ho un messaggio che dice 
"URL di rete locale non valido
Hai configurato un certificato HTTPS in Home Assistant. Ciò significa che il tuo URL interno deve essere impostato su un dominio coperto dal certificato." 
Ora va a capire dove ho sbagliato

**Reply by Scialla** [reply]
Hai configurato il certificato con let's encrypt, usando poi duckdns o similari? Se si non puoi più usare l'IP locale in http ma solo in https, e solo da browser dove, attivando un'eccezione, ti consente di andare avanti, da app companion non puoi farlo.
In sostanza hai configurato per accedere in https da remoto e, tramite let's encrpyt, ottieni i certificati che vengono rilasciati per il tuo dominio, non possono essere rilasciati per un ip, quindi ti da errore.
Se vuoi fare una cosa come si deve, per far funzionare anche su ip locale, devi usare un reverse proxy, guardati la guida per caddy su inDomus, oppure passa direttamente a creare un tunnel, guardati la guida con cloudflare

**Reply by fraygon** [reply]
In effetti è come dici tu. Non ho il reverse proxy. Posso installarlo con duck e let's già installati o devo rifare tutto da zero?

**Reply by Scialla** [reply]
Devi togliere let's encrypt e installare caddy e lasciare che i certificati se li gestisca lui, leggi la guida che spiega tutto

**Reply by fraygon** [reply]
Se posso alcune domande. Sulla guida non ho letto la parte di eliminare let's encrypt quindi io lo elimino e tolgo anche i file in configuration o elimino solo il componente aggiuntivo?
 `http:
  ssl_certificate: /ssl/fullchain.pem
  ssl_key: /ssl/privkey.pem
  use_x_forwarded_for: true
  trusted_proxies:
    - 192.xxxxxx
    - 172.xxxxx.0/xx
  ip_ban_enabled: true
  login_attempts_threshold: 5`
Il mio modem attualmente ha queste porte 8123 e 443, l'esempio parla di 9000. Qui sono un pò in confusione.

**Reply by Scialla** [reply]
Intanto bisognerebbe capire se usi HAOS, e quindi fai tutto con gli addon, o se usi docker o altro... nel primo caso io non ho mai usato HAOS e quindi ti dico di affidarti alla guida..
Comunque devi togliere la riga di let's encrypt dalla configurazione.
Il discorso della porta è un esempio... tu da fuori usi la porta che vuoi, la apri sul router, la indirizzi all'IP di caddy, poi sulla configurazione imposti che la chiamata in arrivo, ad esempio, sul tuo fqdn sulla porta 9000, la giri (redirect) all'IP di HA sulla porta 8123... segui la guida che non è difficile

**Reply by fraygon** [reply]
Uso HAOS su RPI4. Grazie dei suggerimenti. domani provo e spero di non bloccare tutto. Ci sono già passato e non è piacevole. Grazie

**Reply by fraygon** [reply]
Come immaginavo ho fatto un macello. Ho disinstallato let's encrypt, ho installato caddy2 base, come suggerito dalla guida inDomus, ho seguito anche la guida "Aggiornare il proprio record su DuckDNS tramite automazione Home Assistant" e vedo correttamente le entità. Il mio codice in configuration è il seguente
`http:
  ssl_certificate: /ssl/fullchain.pem
  ssl_key: /ssl/privkey.pem
  use_x_forwarded_for: true
  trusted_proxies:
    - 127.0.0.1   # loopback ipv4
    - ::1  # loopback ipv6
    - 172.30.33.0/24  
    - 192.168.1.0/24  
` Se tolgo le righe   
 ssl_certificate: /ssl/fullchain.pem
  ssl_key: /ssl/privkey.pem
non accedo nè da duck nè in locale. se le scrivo accedo da duck e tramite https://192.168.1.132:8123/ ma ho sempre il messaggio "URL di rete locale non valido".
Sul modem ho configurato port mapping uno con porte 9000 - 8123.
Mi sono spulciato una montagna di post su caddy e alla fine ho fatto più casino che altro. Mi potete aiutare?

**Reply by Scialla** [reply]
tanto per cominciare, se vuoi usare la TLS-ALPN Challenge per ottenere i certificati, devi mappare sul modem la porta 443 in ingresso che vada in uscita sulla 443 dell'IP dove gira HAOS. Se invece vuoi usare la HTTP Challenge devi fare la stessa cosa ma con la porta 80, come è scritto chiaramente in guida, ma sembra che tu non lo abbia fatto...
Poi non ha senso fare come hai fatto, ovvero mappare, sul modem, la 9000 verso la 8123, quello lo fai quando imposti i parametri del reverse proxy, come descritto chiaramente in guida.... sul modem mappi la 9000 verso la 9000 sull'ip dove gira HAOS.

In sostanza devi seguire la GUIDA che spiega tutto per filo e per segno, senza dimenticarti di togliere le due righe dal configuration.yaml e di aggiungere:
```
http:
  use_x_forwarded_for: true
  trusted_proxies: 
    - XXX.XXX.XXX.XXX
```
dove “XXX.XXX.XXX.XXX” rappresenterà l’indirizzo IP del reverse proxy (o la classe IP), il quale può essere:

127.0.0.1, se Caddy o il reverse proxy è installato sulla stesso computer di Home Assistant come servizio;
172.0.0.0/8 (solitamente*), se Caddy è installato su Docker sulla stesso computer di Home Assistant (anch’esso installato su Docker);
l’indirizzo IP LAN di Caddy o del reverse proxy, se installato su un computer diverso da quello di Home Assistant.

Come riportato nella GUIDA da cui ho fatto copia&incolla

Non saprei come altro aiutarti oltre a consigliarti di seguire esattamente quando indicato nella guida, anche perché, se ti limiti a dire che "non funziona" è difficile capire cosa hai sbagliato...

**Reply by fraygon** [reply]
Dopo tanto penare ci sono riuscito anche grazie al notevole aiuto di chatGTP. Il casino più grosso era sulle porte del schifosissimo modem fastweb. Ora mi funziona HTTP e quando manca la rete da fuori riesco finalmente a vedere H.A. sul pc e sul tablet, sul telefono ancora no ma non demordo. Grazie per l'aiuto
